modules/nf-core/gmmdemux/meta.yml

```diff
@@ -0,0 +1,110 @@
+---
+# yaml-language-server: $schema=https://raw.githubusercontent.com/nf-core/modules/master/modules/meta-schema.json
+name: "gmmdemux"
+
+description: GMM-Demux is a Gaussian-Mixture-Model-based software for processing sample barcoding data (cell hashing and MULTI-seq).
+keywords:
+  - demultiplexing
+  - hashing-based deconvoltion
+  - single-cell
+tools:
+  - "gmmdemux":
+      description: "GMM-Demux is a Gaussian-Mixture-Model-based software for processing sample barcoding data (cell hashing and MULTI-seq)."
+      homepage: "https://pypi.org/project/GMM-Demux/"
+      documentation: "https://github.com/CHPGenetics/GMM-Demux"
+      tool_dev_url: "https://github.com/CHPGenetics/GMM-demux"
+      doi: "10.1186/s13059-020-02084-2"
+      licence: ["MIT"]
+
+input:
+  # Only when we have meta
+  - meta:
+      type: map
+      description: |
+        Groovy Map containing sample information
+        e.g. `[ id:'sample1', single_end:false ]`
+  - cell_hashing_matrix:
+      type: file
+      description: path to file containing matrix from cell hashing data, the tool receives either CSV files or TSV, type must be specified using parameters
```

Comment:
If the typue MUST be specified using parameters, then this is OK to have as an input cahnnel. Everythin else via `ext.args` (as I've said a few times above :grimacing: )

Reply:
Yep, in this case this are the input files, originally the tool gets only one path to the directory where these 3 files are stored.  
nf-core cannot receive the path to the directory stored in test-datasets, that's why we need the 3 paths to create an intermediate folder which is later given as input

Reply:
If the tool is noramlly meant to accept a directory:

1. Package the three test files into a directory and gzip it
2. Upload to test-datasets
3. Add to the test a 'setup' block where you specify the gzip archive and use the GUNZIP module to extract it
4. Pass GUNZIP.out.archive (or w/e it is) as the (single) input to the module